components/post_view/post_attachment_opengraph/post_attachment_opengraph.jsx

```diff
@@ -37,111 +38,71 @@ export default class PostAttachmentOpenGraph extends React.PureComponent {
 
         isEmbedVisible: PropTypes.bool,
         toggleEmbedVisibility: PropTypes.func.isRequired,
-
-        actions: PropTypes.shape({
-
-            /**
-             * The function to get open graph data for a link
-             */
-            getOpenGraphMetadata: PropTypes.func.isRequired,
-        }).isRequired,
     }
 
     constructor(props) {
         super(props);
-        this.largeImageMinWidth = 150;
-        this.imageDimentions = { // Image dimentions in pixels.
-            height: 80,
-            width: 80,
-        };
-        this.textMaxLength = 300;
-        this.textEllipsis = '...';
-        this.largeImageMinRatio = 16 / 9;
-        this.smallImageContainerLeftPadding = 15;
 
-        this.imageRatio = null;
-
-        this.smallImageContainer = null;
-        this.smallImageElement = null;
-
-        this.IMAGE_LOADED = {
-            LOADING: 'loading',
-            YES: 'yes',
-            ERROR: 'error',
-        };
-
-        this.fetchData = this.fetchData.bind(this);
-        this.onImageLoad = this.onImageLoad.bind(this);
-        this.onImageError = this.onImageError.bind(this);
         this.handleRemovePreview = this.handleRemovePreview.bind(this);
-    }
 
-    UNSAFE_componentWillMount() { // eslint-disable-line camelcase
-        const removePreview = this.isRemovePreview(this.props.post, this.props.currentUser);
-
-        this.setState({
-            imageLoaded: this.IMAGE_LOADED.LOADING,
-            hasLargeImage: false,
+        const removePreview = this.isRemovePreview(props.post, props.currentUser);
+        const imageUrl = this.getBestImageUrl(props.openGraphData);
+        const hasLargeImage = this.hasLargeImage(props.post.metadata.images[imageUrl]);
```

Comment:
do we need the whole **post** as a prop or just with the `imageMetadata` is enough ?

Reply:
We are not adding the `post` prop here in the PR. It is part of the props to the component and it is being used in couple of functions around in the component.